my/coding/commits.py

```diff
@@ -1,38 +1,64 @@
 """
 Git commits data for repositories on your filesystem
 """
+REQUIRES = [
+    'gitpython',
+]
+
 
 import shutil
 import string
 from pathlib import Path
 from datetime import datetime, timezone
-from typing import List, NamedTuple, Optional, Dict, Any, Iterator, Set
+from dataclasses import dataclass, field
+import functools
+from typing import List, NamedTuple, Optional, Iterator, Set, Sequence
+
+
+from my.core import PathIsh, LazyLogger, make_config
+from my.core.cachew import cache_dir
+from my.core.common import mcachew
+from my.core.warnings import high
+
+
+from my.config import commits as user_config
+@dataclass
+class commits_cfg(user_config):
+    roots: Sequence[PathIsh] = field(default_factory=list)
```

Comment:
Interesting that without it, it would complain
`ValueError: mutable default <class 'list'> for field roots is not allowed: use default_factory` -- so would require you to use `tuple` in the user config. Didn't know about that.. but good that relatively easy to workaround with this field trick.. Not sure if there is an easier way (maybe some `@dataclass` annotation attributes).